### PR TITLE
Fix the issue when cross-building for ARMv8

### PR DESCRIPTION
**Pull request by chendave**
Signed-off-by: Dave Chen <dave.chen@arm.com>

**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

```make crossbuild WHAT=edgecore GOARM=GOARM8```
will hit below exception like this, 
```Invalid GOARM value. Must be 5, 6, or 7.```

should clear the env when building the binary for armv8

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

@fisherxu

### CODE REVIEW

File: hack/lib/golang.sh [hunk 1]
```diff
@@ -294,6 +294,7 @@ kubeedge::golang::cross_build_place_binaries() {
     local name="${bin##*/}"
     if [ "${goarm}" == "8" ]; then
       set -x
+      GOARM="" # need to clear the value since golang compiler doesn't allow this env when building the binary for ARMv8.
       GOARCH=arm64 GOOS="linux" CGO_ENABLED=1 CC=aarch64-linux-gnu-gcc go build -o ${KUBEEDGE_OUTPUT_BINPATH}/${name} -ldflags "$ldflags" $bin
```
Comment (gy95) [inline hack/lib/golang.sh:298]: we could specify `GOARM=""` just like set `GOARM=${goarm}` specifically for GOARM7 .
ref: https://github.com/golang/go/wiki/GoArm we should not set GOARM for GOARM8

**Review by gy95 — COMMENTED** [review]
LGTM

File: hack/lib/golang.sh (hunk 1 above)
Comment (chendave) [reply hack/lib/golang.sh:298]: yes, that is why the command like this will fail, 
```make crossbuild WHAT=edgecore GOARM=GOARM8```

Comment (fisherxu) [reply hack/lib/golang.sh:298]: make sense, in golang `GOARM=GOARM8` will failed.. I'm wondering whether it's reasonable ;-)

**Review by fisherxu — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**